aiochsa/types.py

```diff
@@ -173,46 +199,59 @@ def from_json(self, value: str) -> datetime:
             return result.replace(tzinfo=self._tzinfo).astimezone(timezone.utc)
 
 
-class UUIDType(BaseType):
+class UUIDType(BaseType[UUID, str]):
     py_type = UUID
 
     @classmethod
     def escape(cls, value: UUID, escape=None) -> str:
         return f"'{value}'"
 
     @classmethod
-    def to_json(cls, value: UUID, to_json: Callable) -> JsonType:
+    def to_json(cls, value: UUID, to_json: Callable) -> str:
         return str(value)
 
+    def from_json(self, value: str) -> UUID:
+        return self.py_type(value)
+
 
-class IPv4Type(BaseType):
+class IPv4Type(BaseType[IPv4Address, str]):
     py_type = IPv4Address
 
     @classmethod
-    def to_json(cls, value: IPv4Address, to_json: Callable) -> JsonType:
+    def to_json(cls, value: IPv4Address, to_json: Callable) -> str:
         return str(value)
 
+    def from_json(self, value: str) -> IPv4Address:
+        return self.py_type(value)
+
 
-class IPv6Type(BaseType):
+class IPv6Type(BaseType[IPv6Address, str]):
     py_type = IPv6Address
 
     @classmethod
-    def to_json(cls, value: IPv6Address, to_json: Callable) -> JsonType:
+    def to_json(cls, value: IPv6Address, to_json: Callable) -> str:
         return str(value)
 
+    def from_json(self, value: str) -> IPv6Address:
+        return self.py_type(value)
+
 
-class NothingType(BaseType):
-    py_type = NoneType
+class NothingType(BaseType[None, None]):
+    py_type = NoneType  # type: ignore
 
-    def escape(self, value: NoneType, escape=None) -> str:
+    def escape(self, value: None, escape=None) -> str:  # type: ignore
```

Comment:
This case drives me mad. Any idea why mypy complaints with `Signature of "escape" incompatible with supertype "BaseType"`?

Reply:
In `BaseType` it's annotaded with `escape: Callable`.
Write `Optional[Callable]` there for example.

Reply:
Both variants doesn't solve problem.

Reply:
Just got it, the problem was in `@classmethod`